File: back/models/Promotion.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const PromotionSchema = new Schema({
  name: {
    type: String,
    required: true,
  },
  year: {
    type: Number,
    required: true,
  },
  students: [
    {
      type: Schema.Types.ObjectId,
      ref: "Student",
    },
  ],
  teachers: [
    {
      type: Schema.Types.ObjectId,
      ref: "Teacher",
    },
  ],
  lessons: [
    {
      type: Schema.Types.ObjectId,
      ref: "Lesson",
    },
  ],
});

module.exports = mongoose.model("Promotion", PromotionSchema);

File: h1/lib/models/promotion.dart
import 'package:hive/hive.dart';
import 'lesson.dart';
import 'student.dart';

part 'promotion.g.dart';

@HiveType(typeId: 4)
class Promotion {
  @HiveField(0)
  final String id;

  @HiveField(1)
  final String name;

  @HiveField(2)
  final int year;

  @HiveField(3)
  final List<Lesson> lessons;

  @HiveField(4)
  final List<Student> students;

  Promotion({
    required this.id,
    required this.name,
    required this.year,
    required this.lessons,
    required this.students,
  });
}
